### PR TITLE
Implementation of checkList() and displayQueryList() in VField

### PR DESCRIPTION
**Pull request by ZMarouani**

### CODE REVIEW

File: galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt [hunk 1]
```diff
@@ -1520,85 +1508,204 @@ abstract class VField protected constructor(width: Int, height: Int) : VConstant
...
+                }
+              } catch (e: Throwable) {
+                throw VRuntimeException(e)
+              }
+              result
+            } else {
+              lines[0][selected]
+            }
+          } as String?
+
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/form/VField.kt:1704]: check kopi this is displayQueryList method so i think that you need to change colbuf to a list of column then create qrybuf and pass it to displayQueryList

File: galite-core/src/main/kotlin/org/kopi/galite/list/VStringColumn.kt [hunk 2]
```diff
@@ -20,7 +20,7 @@ package org.kopi.galite.list
 
 import kotlin.reflect.KClass
 
-class VStringColumn(title: String,
+class VStringColumn(title: String?,
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/list/VStringColumn.kt:23]: undo

File: galite-core/src/main/kotlin/org/kopi/galite/list/VListColumn.kt [hunk 3]
```diff
@@ -18,18 +18,29 @@
...
 package org.kopi.galite.list
 
+import org.jetbrains.exposed.sql.Column
 import kotlin.reflect.KClass
 
 import org.kopi.galite.l10n.ListLocalizer
 
 abstract class VListColumn(
-  var title: String,
+  var title: String?,
```
Comment (h-haddad) [inline galite-core/src/main/kotlin/org/kopi/galite/list/VListColumn.kt:27]: undo

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VFieldTests.kt [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+import org.kopi.galite.form.VFieldException
+import org.kopi.galite.form.VPosition
+import org.kopi.galite.list.VColumn
+import org.kopi.galite.list.VList
+import org.kopi.galite.list.VListColumn
+import org.kopi.galite.list.VStringColumn
+import org.kopi.galite.tests.JApplicationTestBase
+import org.kopi.galite.visual.MessageCode
+import org.kopi.galite.visual.VCommand
+import kotlin.test.assertFailsWith
```
Comment (h-haddad) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VFieldTests.kt:33]: reorder

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VFieldTests.kt [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+  @Test
+  fun getCheckListEmptyFieldTest() {
+    assertFailsWith<VFieldException>(MessageCode.getMessage("VIS-00001")) {
+      transaction {
+        //checks if it exists in database ( at least one element )
+        ageField.checkList()
+      }
+    }
+  }
```
Comment (h-haddad) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VFieldTests.kt:47]: remove

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VFieldTests.kt [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+
+  @Test
+  fun getCheckListVStringFieldTest1() {
```
Comment (h-haddad) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VFieldTests.kt:50]: change name

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VFieldTests.kt [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+        nameField.checkList()
+      }
+    }
+  }
+
+  @Test
+  fun getCheckListVStringFieldTest2() {
+     object {
+      init {
+        val userTable = User
```
Comment (h-haddad) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VFieldTests.kt:62]: remove

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VFieldTests.kt [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+          userTable.insert {
+            it[id] = 2
+            it[name] = "Fabienne BUGHIN"
+          }
+          userTable.insert {
+            it[id] = 3
+            it[name] = "FABIENNE BUGHIN2"
+          }
+        }
+      }
```
Comment (h-haddad) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VFieldTests.kt:78]: move this code in second transaction and delete SchemaUtils.create(User)

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VFieldTests.kt [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+    }
+    //CheckList function works without exceptions in this case .
+      transaction {
+        nameField.checkList()
+      }
+  }
+
+@Test
+  fun getCheckListVStringFieldTest3() {
+     object {
```
Comment (h-haddad) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VFieldTests.kt:88]: delete object

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VFieldTests.kt [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+            it[name] = "AUDREY"
+          }
+          userTable.insert {
+            it[id] = 2
+            it[name] = "Fabienne BUGHIN"
+          }
+          userTable.insert {
+            it[id] = 3
+            it[name] = "FABIENNE BUGHIN2"
+          }
```
Comment (h-haddad) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VFieldTests.kt:108]: move this code in second trnsaction and remove create(user)

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VFieldTests.kt [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+
+  companion object{
+    init {
+      FormSample.tb1.id.value = 1
+      FormSample.tb1.name.value = "AUDREY"
+      transaction {
+        SchemaUtils.create(User)
+      }
+    }
+  }
```
Comment (h-haddad) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VFieldTests.kt:128]: remove companion object{ and move code in the second init

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VFieldTests.kt [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+            it[name] = "Fabienne BUGHIN"
+          }
+          userTable.insert {
+            it[id] = 3
+            it[name] = "FABIENNE BUGHIN2"
+          }
+        }
+      }
+    }
+    //CheckList function works without exceptions in this case .
```
Comment (h-haddad) [inline galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VFieldTests.kt:80]: assert the test

**Review by h-haddad — CHANGES_REQUESTED** [review]

File: galite-core/src/main/kotlin/org/kopi/galite/list/VListColumn.kt (hunk 3 above)
Comment (ZMarouani) [reply galite-core/src/main/kotlin/org/kopi/galite/list/VListColumn.kt:27]: title is nullable

File: galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VFieldTests.kt (hunk 12 above)
Comment (ZMarouani) [reply galite-tests/src/test/kotlin/org/kopi/galite/tests/form/VFieldTests.kt:80]: in this case , the function works proparly and have no uotput neither throws exception , in this case the test doesn't need assertion , it just passes if the function works

File: galite-core/src/main/kotlin/org/kopi/galite/list/VStringColumn.kt (hunk 2 above)
Comment (ZMarouani) [reply galite-core/src/main/kotlin/org/kopi/galite/list/VStringColumn.kt:23]: title can be null

**Review by h-haddad — CHANGES_REQUESTED** [review]

**Pull request closed without merging**